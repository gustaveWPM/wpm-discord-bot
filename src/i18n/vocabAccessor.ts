import type { Locales } from './typesafe-i18n/i18n-types';

import L from './typesafe-i18n/i18n-node';
import { getCurrentLocale } from './ctx';

export const vocabAccessor = (locale: Locales = getCurrentLocale()) => L[locale];
