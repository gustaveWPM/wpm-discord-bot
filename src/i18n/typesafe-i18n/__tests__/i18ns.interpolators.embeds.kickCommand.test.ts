import { beforeAll, describe, expect, test } from 'bun:test';

import type { Locales } from '../i18n-types';

import { loadAllLocalesAsync } from '../i18n-util.async';
import { vocabAccessor } from '../../vocabAccessor';
import { locales } from '../i18n-util';

const needle = '__TEST_NEEDLE__';

const scope = (locale: Locales) => vocabAccessor(locale).embeds.kickCommand;

describe('I18n interpolators (embeds > kickCommand)', () => {
  beforeAll(async () => await loadAllLocalesAsync());

  test('kicksAmountOverLimit', () => {
    for (const locale of locales) {
      const item = scope(locale).kicksAmountOverLimit({ limit: needle });

      expect(item.includes(needle)).toBe(true);
    }
  });

  test('kicksAmount', () => {
    for (const locale of locales) {
      const item = scope(locale).kicksAmount({ kicksAmount: needle });

      expect(item.includes(needle)).toBe(true);
    }
  });

  test('kicksAmount > 0', () => {
    const items: string[] = [];
    for (const locale of locales) {
      items.push(scope(locale).kicksAmount({ kicksAmount: 0 }));
    }

    expect(items).toMatchSnapshot();
  });

  test('kicksAmount > singular', () => {
    const items: string[] = [];
    for (const locale of locales) {
      items.push(scope(locale).kicksAmount({ kicksAmount: 1 }));
    }

    expect(items).toMatchSnapshot();
  });

  test('kicksAmount > plural', () => {
    const items: string[] = [];
    for (const locale of locales) {
      items.push(scope(locale).kicksAmount({ kicksAmount: 2 }));
    }

    expect(items).toMatchSnapshot();
  });
});
