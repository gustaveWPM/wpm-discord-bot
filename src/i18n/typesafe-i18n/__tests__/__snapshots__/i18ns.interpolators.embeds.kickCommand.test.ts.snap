// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`I18n interpolators (embeds > kickCommand) kicksAmount > 0 1`] = `
[
  "You've been kicked 0 times.",
  "Vous avez été exclus 0 fois.",
]
`;

exports[`I18n interpolators (embeds > kickCommand) kicksAmount > singular 1`] = `
[
  "You've been kicked 1 time.",
  "Vous avez été exclus 1 fois.",
]
`;

exports[`I18n interpolators (embeds > kickCommand) kicksAmount > plural 1`] = `
[
  "You've been kicked 2 times.",
  "Vous avez été exclus 2 fois.",
]
`;
